RxSwiftExample/Providers/Profile/ProfileListProvider.swift

```diff
@@ -0,0 +1,56 @@
+//
+//  ProfileListProvider.swift
+//  RxSwiftExample
+//
+//  Created by Johnny Yen on 2020/6/3.
+//  Copyright © 2020 Test. All rights reserved.
+//
+
+import Foundation
+import RxSwift
+import CoreData
+
+class ProfileListProvider {
+    private var list: [ProfileProvider] = []
+    private let storageService: StorageService
+
+    //Outputs
+    let providers = BehaviorSubject<[ProfileProvider]>(value: [])
+
+    init(_ storageService: StorageService) {
+        self.storageService = storageService
+    }
+
+    deinit {
+        providers.onCompleted()
+    }
+
+    func update() -> Observable<Void> {
+        let storServ = storageService
+        return storServ.fetchAll()
+            .map { [ProfileProvider]($0, storServ) }
+            .map { [weak self] in
+                self?.list = $0
+                self?.providers.onNext($0)
+            }
+    }
+
+    func clearAll() -> Observable<Void> {
+        let storServ = storageService
+//        return storServ.deleteAll()
+        return Observable.error(StorageService.Errors.deleteAll)
+            .flatMap { storServ.fetchAll() }
+            .map { [ProfileProvider]($0, storServ) }
+            .map { [weak self] in
+                self?.list = $0
+                self?.providers.onNext($0)
+            }
+    }
+}
+
+extension Array where Element == ProfileProvider {
+
+    init(_ profiles: [Profile], _ storageServices: StorageService) {
+        self = profiles.map{ ProfileProvider(storageServices, $0) }
+    }
+}
```

Comment:
It seems to be an extension that belongs to **ProfileProvier**, maybe put it to the **ProfileProvide.swift** or create a standalone file with it.